pex/tools/commands/venv.py

```diff
@@ -61,12 +63,206 @@ def _copytree(
+        bin_path = os.environ.get("PEX_VENV_BIN_PATH", {bin_path!r})
+        if bin_path != "false":
+            PATH = os.environ.get("PATH", "").split(os.pathsep)
+            if bin_path == "prepend":
+                PATH.insert(0, bin_dir)
+            elif bin_path == "append":
+                PATH.append(bin_dir)
+            else:
+                sys.stderr.write(
+                    "PEX_VENV_BIN_PATH must be one of 'false', 'prepend' or 'append', given: "
+                    "{{!r}}\\n".format(
+                        bin_path
                     )
-                    if not collisions_ok:
-                        raise e
+                )
+                sys.exit(1)
+            os.environ["PATH"] = os.pathsep.join(PATH)
+
+        PEX_EXEC_OVERRIDE_KEYS = ("PEX_INTERPRETER", "PEX_SCRIPT", "PEX_MODULE")
+        pex_overrides = dict(
+            (key, os.environ.pop(key)) for key in PEX_EXEC_OVERRIDE_KEYS if key in os.environ
+        )
+        if len(pex_overrides) > 1:
+            sys.stderr.write(
+                "Can only specify one of {{overrides}}; found: {{found}}\\n".format(
+                    overrides=", ".join(PEX_EXEC_OVERRIDE_KEYS),
+                    found=" ".join("{{}}={{}}".format(k, v) for k, v in pex_overrides.items())
+                )
+            )
+            sys.exit(1)
+
+        pex_script = pex_overrides.get("PEX_SCRIPT")
+        if pex_script:
+            script_path = os.path.join(bin_dir, pex_script)
+            os.execv(script_path, [script_path] + sys.argv[1:])
+
+        pex_interpreter = pex_overrides.get("PEX_INTERPRETER", "").lower() in ("1", "true")
+        PEX_INTERPRETER_ENTRYPOINT = "code:interact"
+        entry_point = (
+            PEX_INTERPRETER_ENTRYPOINT
+            if pex_interpreter
+            else pex_overrides.get("PEX_MODULE", {entry_point!r} or PEX_INTERPRETER_ENTRYPOINT)
+        )
+        if entry_point == PEX_INTERPRETER_ENTRYPOINT and len(sys.argv) > 1:
+            args = sys.argv[1:]
+            arg = args[0]
+            if arg == "-m":
+                if len(args) < 2:
+                    sys.stderr.write("Argument expected for the -m option\\n")
+                    sys.exit(2)
+                entry_point = module = args[1]
+                sys.argv = args[1:]
+                # Fall through to entry_point handling below.
+            else:
+                filename = arg
+                sys.argv = args
+                if arg == "-c":
+                    if len(args) < 2:
+                        sys.stderr.write("Argument expected for the -c option\\n")
+                        sys.exit(2)
+                    filename = "-c <cmd>"
+                    content = args[1]
+                    sys.argv = ["-c"] + args[2:]
+                elif arg == "-":
+                    content = sys.stdin.read()
+                else:
+                    with open(arg) as fp:
+                        content = fp.read()
```

Comment:
You may want to open this with `rb`. From the docs https://docs.python.org/3/library/functions.html#compile:

> source can either be a normal string, a byte string, or an AST object.

Why do it? I found in https://github.com/pantsbuild/pants/pull/11213/commits/d8b50388d12e204e953dde50d6fa6c2d127a0bb0 for Pants' dep inference that opening with `rb` avoided Unicode errors.

Reply:
The relevant doc is here and different: https://docs.python.org/2.7/library/functions.html#compile